GameServer.cpp

```diff
@@ -404,7 +404,7 @@ void GameServer::DoTimedUpdates()
 	// Get the current time
 	currentTime = time(0);
 	// Check for timed out game entries
-	for (i = numGames - 1; i >= 0; --i)
+	for (std::size_t i = numGames - 1; i != (std::numeric_limits<std::size_t>::max)(); --i)

```

Comment:
I'm pretty sure this logic is correct for a non-signed type. @DanRStevens, I wanted you to review the loop conditions closely.

I am able to run a local instance on my computer and it appears to work properly (Not cause an infinite loop with this change).

Reply:
It may be correct, but there is a less confusing idiom that can be used here:
```cpp
for (std::size_t i = numGames; i-- > 0; )
```

The `i--` expression returns the value of `i` before being decremented. Hence, if `i == 0`, the loop will exit. In particular, if `numGames == 0`, the loop will be skipped entirely. Once the check is complete, the result of decrement `i--` is seen, so the loop will run with the decremented value of `i`. The loop will execute with `i` in the range `0 .. (numGames - 1)` within the loop body.

Reply:
Ah, I had forgotten about that syntax. Thanks for pointing it out. I still feel like there should be an easier to read decrementing loop construct...

Reply:
Just remember the "goes to" operator :wink:
https://stackoverflow.com/questions/1642028/what-is-the-operator-in-c

----

On a more serious note, hopefully the Range support being added to C++20 will provide better alternatives.